src/corona_hakab_model/infection.py

```diff
@@ -42,14 +48,31 @@ def _perform_infection(self) -> List[Agent]:
         # calculate the infections boolean vector
 
         infections = self.manager.susceptible_vector & (np.random.random(u.shape) < u)
-        infected_indices = np.flatnonzero(infections)
+        return infections
+
+    def _infect_random_connections(self):
+        connections = self.manager.num_of_random_connections * self.manager.random_connections_factor
+        probs_not_infected_from_connection = np.ones_like(connections, dtype=float)
+        for connection_type in connection_types.With_Random_Connections:
+            for circle in self.manager.social_circles_by_connection_type[connection_type]:
+                agents_id = [a.index for a in circle.agents]
+
+                if len(agents_id) == 1:
+                    # One-Agent circle, you can't randomly meet yourself..
+                    continue
+
+                total_infectious_random_connections = np.dot(
+                    self.manager.contagiousness_vector[(agents_id,)],
+                    connections[(agents_id, [connection_type] * len(agents_id))],
+                )
+
+                prob = total_infectious_random_connections / circle.total_random_connections
 
-        # new_infected: dict -
-        # key = medical state (currently only susceptible state which an agent can be infected)
-        # value = list of agents
-        new_infected = list()
-        for index in infected_indices:
-            agent = self.manager.agents[index]
-            new_infected.append(agent)
+                probs_not_infected_from_connection[(agents_id, [connection_type] * len(agents_id))] = \
+                    1 - prob * self.manager.random_connections_strength[connection_type]
```

Comment:
You do not need the 1 here, - is good enough :)

Reply:
what? why?